CompuCell3D/core/CompuCell3D/steppables/PDESolvers/ReactionDiffusionSolverFE.cpp

```diff
@@ -813,11 +813,11 @@ void ReactionDiffusionSolverFE::step(const unsigned int _currentStep) {
             for (int idx = 0; idx < numberOfFields; ++idx) {
                 prepCellTypeField(idx); // here we initialize celltype array  boundaries - we do it once per  MCS
                 boundaryConditionInit(idx);//initializing boundary conditions
-                solveRDEquationsSingleField(idx); //reaction-diffusion
                 //secretion
                 for (unsigned int j = 0; j < diffSecrFieldTuppleVec[idx].secrData.secretionFcnPtrVec.size(); ++j) {
                     (this->*diffSecrFieldTuppleVec[idx].secrData.secretionFcnPtrVec[j])(idx);
                 }
+				solveRDEquationsSingleField(idx); //reaction-diffusion

```

Comment:
OK, one more request/ question to consider. Shouldn't we firs do secretion in all fields and then do the loop to solveRD equations? Something like this:

for(all fields):
   perform secretion

for (all fields)
   solveRDEquationSingleField

I think this would be correct order of operations . 

Otherwise during solveRDEquationsSingleField call for some fields we would be using field values with secretion accounted for and some field values would be using values before secretion took place

Reply:
Agreed.